diff --git a/setup.py b/setup.py
index 388e629..5eb54b4 100644
--- a/setup.py
+++ b/setup.py
@@ -243,15 +243,8 @@ def cython_extensions_and_necessity():
                   for name in CYTHON_EXTENSION_MODULE_NAMES]
   config = os.environ.get('CONFIG', 'opt')
   prefix = 'libs/' + config + '/'
-  if "darwin" in sys.platform:
-    extra_objects = [prefix + 'libares.a',
-                     prefix + 'libboringssl.a',
-                     prefix + 'libgpr.a',
-                     prefix + 'libgrpc.a']
-    core_c_files = []
-  else:
-    core_c_files = list(CORE_C_FILES)
-    extra_objects = []
+  core_c_files = list(CORE_C_FILES)
+  extra_objects = []
   extensions = [
       _extension.Extension(
           name=module_name,
diff --git a/src/python/grpcio/commands.py b/src/python/grpcio/commands.py
index 0a30971..44b1eff 100644
--- a/src/python/grpcio/commands.py
+++ b/src/python/grpcio/commands.py
@@ -57,7 +57,7 @@ Glossary
 .. glossary::
 
   metadatum
-    A key-value pair included in the HTTP header.  It is a 
+    A key-value pair included in the HTTP header.  It is a
     2-tuple where the first entry is the key and the
     second is the value, i.e. (key, value).  The metadata key is an ASCII str,
     and must be a valid HTTP header name.  The metadata value can be
@@ -253,38 +253,6 @@ class BuildExt(build_ext.build_ext):
     LINK_OPTIONS = {}
 
     def build_extensions(self):
-        if "darwin" in sys.platform:
-            config = os.environ.get('CONFIG', 'opt')
-            target_path = os.path.abspath(
-                os.path.join(
-                    os.path.dirname(os.path.realpath(__file__)), '..', '..',
-                    '..', 'libs', config))
-            targets = [
-                os.path.join(target_path, 'libboringssl.a'),
-                os.path.join(target_path, 'libares.a'),
-                os.path.join(target_path, 'libgpr.a'),
-                os.path.join(target_path, 'libgrpc.a')
-            ]
-            # Running make separately for Mac means we lose all
-            # Extension.define_macros configured in setup.py. Re-add the macro
-            # for gRPC Core's fork handlers.
-            # TODO(ericgribkoff) Decide what to do about the other missing core
-            #   macros, including GRPC_ENABLE_FORK_SUPPORT, which defaults to 1
-            #   on Linux but remains unset on Mac.
-            extra_defines = [
-                'EXTRA_DEFINES="GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK=1"'
-            ]
-            make_process = subprocess.Popen(
-                ['make'] + extra_defines + targets,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE)
-            make_out, make_err = make_process.communicate()
-            if make_out and make_process.returncode != 0:
-                sys.stdout.write(str(make_out) + '\n')
-            if make_err:
-                sys.stderr.write(str(make_err) + '\n')
-            if make_process.returncode != 0:
-                raise Exception("make command failed!")
 
         compiler = self.compiler.compiler_type
         if compiler in BuildExt.C_OPTIONS:
